components/dialog/index.css

```diff
@@ -11,377 +11,429 @@ governing permissions and limitations under the License.
+	font-size: var(--mod-standard-dialog-heading-font-size, var(--spectrum-standard-dialog-heading-font-size));
+	font-weight: var(--mod-standard-dialog-heading-font-weight, var(--spectrum-standard-dialog-heading-font-weight));
+	font-style: var(--spectrum-standard-dialog-heading-font-style);
+	font-family: var(--spectrum-standard-dialog-heading-font-family);
+	line-height: var(--mod-standard-dialog-heading-line-height, var(--spectrum-standard-dialog-heading-line-height));
+	color: var(--mod-standard-dialog-heading-text-color, var(--spectrum-standard-dialog-heading-text-color));
+	outline: none; /* Hide focus outline */
+
+	&.spectrum-Dialog-heading--noHeader {
+		grid-area: heading-start / heading-start / header-end / header-end;
+		padding-inline-end: 0;
+	}
+
+	/* Remove the default margin on the Typography in the heading */
+	.spectrum-Heading {
+		margin-block-start: 0;
+	}
 }
 
 .spectrum-Dialog-header {
-  grid-area: header;
-  display: flex;
+	grid-area: header;
+	display: flex;
+	inline-size: fit-content;
+	max-inline-size: 100%;
+	margin-block-end: var(--spectrum-standard-dialog-spacing-title-to-description);
 
-  /* Without this,  buttons will be stretched */
-  align-items: center;
-  justify-content: flex-end;
+	/* Without this, buttons will be stretched */
+	align-items: center;
+	justify-content: flex-end;
 
-  box-sizing: border-box;
+	box-sizing: border-box;
 
-  outline: none; /* Hide focus outline around header */
-}
-
-.spectrum-Dialog .spectrum-Dialog-divider {
-  grid-area: divider;
-  inline-size: 100%;
-  margin-block-start: var(--mod-dialog-confirm-divider-block-spacing-end, var(--spectrum-dialog-confirm-divider-block-spacing-end));
-  margin-block-end: var(--mod-dialog-confirm-divider-block-spacing-start, var(--spectrum-dialog-confirm-divider-block-spacing-start));
-}
+	outline: none; /* Hide focus outline around header */
 
-/* remove top vertical padding of spectrum-Dialog-content from bottom margin of first divider after heading */
-.spectrum-Dialog--fullscreen .spectrum-Dialog-heading + .spectrum-Dialog-divider {
-  margin-block-end: calc(var(--mod-dialog-confirm-divider-block-spacing-start, var(--spectrum-dialog-confirm-divider-block-spacing-start)) - (var(--mod-dialog-confirm-description-padding, var(--spectrum-dialog-confirm-description-padding)) * 2));
-}
-
-.spectrum-Dialog--noDivider {
-  .spectrum-Dialog-divider {
-    display: none;
-  }
-
-  .spectrum-Dialog-heading {
-    padding-block-end: calc(
-      var(--mod-dialog-confirm-divider-block-spacing-end, var(--spectrum-dialog-confirm-divider-block-spacing-end)) +
-        var(--mod-dialog-confirm-divider-block-spacing-start, var(--spectrum-dialog-confirm-divider-block-spacing-start)) +
-        var(--mod-dialog-confirm-divider-height, var(--spectrum-dialog-confirm-divider-height))
-    );
-  }
+	/* additional header content should grow/shrink */
+	.spectrum-Body {
+		min-inline-size: fit-content;
+	}
```

Comment:
One thing I haven't figured out is how to get this optional header text to wrap. Any thoughts or ideas? If I make the text long enough, it pushes the title right out of the container. 

I have the same problem with the optional footer content too, actually.